fml/synchronization/sync_switch.cc

```diff
@@ -0,0 +1,41 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/fml/synchronization/sync_switch.h"
+
+namespace fml {
+
+SyncSwitch::Handlers& SyncSwitch::Handlers::SetIfTrue(
+    std::function<void()>&& handler) {
+  true_handler = std::move(handler);
+  return *this;
+}
+
+SyncSwitch::Handlers& SyncSwitch::Handlers::SetIfFalse(
+    std::function<void()>&& handler) {
+  false_handler = std::move(handler);
+  return *this;
+}
+
+SyncSwitch::SyncSwitch() : SyncSwitch(false) {}
+
+SyncSwitch::SyncSwitch(bool value) : value_(value) {}
+
+void SyncSwitch::Execute(const SyncSwitch::Handlers& handlers) {
+  std::scoped_lock guard(mutex_);
```

Comment:
I don't know why GitHub ate my comment about the mutex, so adding it back in:

Please use the shared exclusive mutex in `//flutter/fml/synchronization/shared_mutex.h` with a `SharedLock` here and `UniqueLock` in `SetSwitch`.

Reply:
Discussed offline, scoped_lock makes sense.  It avoids the overhead of a read-write lock when we don't need it and the overhead of a virtual function.